text/trace/0247-peer-service-propagation.md

```diff
@@ -0,0 +1,157 @@
+# Automatic Peer Service Name propagation
+
+Automatic propagation of `peer.service` through `TraceState`.
+
+## Motivation
```

Comment:
The motivation does not read convincing to me. Why does a service need to know who called it? If this is part of a transitive root cause isolation workflow, then you just use distributed traces for that. If this is about some business-specific behavior depending on who called you, e.g. multi-tenant behavior. then I think this mechanism is quite inappropriate - relying on a deployment/infra name of the caller is a pretty narrow use case, not suitable for general purpose multi-tenancy. So please describe Users and Jobs To Be Done of this feature.

Reply:
I could see this being helpful in two scenarios in my work:

1. Sampling rule where you look at a given event and can see that its caller is X. For example, filtering out a noisy branches in a trace but that caller is something you want to keep.
2. Folks who want to get this information eventually via tracing, but can't today, and so if there's an easier way to "add otel" without fully adopting tracing _and_ getting this caller info, that'd be helpful for them.